src/main/java/org/threadly/concurrent/future/SettableListenableFuture.java

```diff
@@ -215,15 +255,20 @@ public void clearResult() {
   }
   
   // should be synchronized on resultLock before calling
-  private void setDone() {
+  private boolean setDone() {
```

Comment:
not sure if we can get this in call cases, but this could probably be achieved through an atomicBoolean instead of needing a lock.